src/Georadix.Core.Tests/Validation/DataAnnotationExtensionsFixture.cs

```diff
@@ -24,43 +45,64 @@ public void AssertValidOnInvalidObjectThrowsArgumentException()
 
             Assert.Equal("sut", ex.ParamName);
             Assert.Contains("Name: Value is required.", ex.Message);
-            Assert.Contains("Title: Value is required.", ex.Message);
         }
 
         [Fact]
-        public void ValidateInvalidObjectReturnsListValidationResults()
+        public void AssertValidOnValidObjectDoesNotThrowException()
         {
-            var sut = new TestModel() { Title = "Title" };
+            var sut = new TestModel
+            {
+                Name = "Name",
+                SubModel = new SubModel
+                {
+                    Property1 = "1"
+                }
+            };
+
+            sut.AssertValid("sut");
+        }
 
+        [Theory]
+        [MemberData("InvalidModelScenarios")]
+        public void ValidateInvalidObjectReturnsListValidationResults(
+            TestModel sut, string[] memberNames, string errorMessage)
+        {
             var validationResults = sut.Validate();
 
             Assert.NotEmpty(validationResults);
 
             var result = validationResults.First();
 
-            Assert.Equal(new string[] { "Name" }, result.MemberNames);
-            Assert.Equal("Value is required.", result.ErrorMessage);
+            Assert.Equal(memberNames, result.MemberNames);
+            Assert.Equal(errorMessage, result.ErrorMessage);
         }
 
         [Fact]
         public void ValidateValidObjectReturnsEmptyListValidationResult()
         {
-            var sut = new TestModel() { Name = "Name", Title = "Title" };
+            var sut = new TestModel() { Name = "Name", SubModel = new SubModel { Property1 = "1" } };
 
             var validationResults = sut.Validate();
 
             Assert.Empty(validationResults);
         }
 
-        private class TestModel
+        public class SubModel
         {
+            [Required(AllowEmptyStrings = false, ErrorMessage = "Value is required.")]
+            public string Property1 { get; set; }
+        }
+
+        public class TestModel
```

Comment:
They had to once they got exposed in the test method as a parameter, it was complaining that the access modified for the class was more restrictive than the method.